src/lib.rs

```diff
@@ -497,6 +478,15 @@ impl<K: EnrKey> Enr<K> {
             return Err(EnrError::UnsupportedIdentityScheme);
         }
 
+        // TODO @age: You can't insert low integers with Enr::insert API now
+        // A solution would be to change the fn signature of ::insert to
+        // `pub fn insert<T: Encodable>(.., value: T)` so that u16 are encoded properly
```

Comment:
I think it makes sense to provide a `insert_encodable` that looks like this and use this instead of calling encode by hand everywhere.

Reply:
Yeah. I think we change the `insert` function to only allow RLP Encodeable objects. The `insert()` function as it stands promotes the wrong behaviour (and prevents correct behaviour as you've pointed out). 

I've forgotten the reason for `insert_raw_rlp()` but I suspect someone is doing RLP encoding outside of the library and may need it, so it can probably stay.